types/nightwatch/index.d.ts

```diff
@@ -1047,13 +1047,13 @@ export interface Expect {
     /**
      * Expect assertions operating on a single element, specified by its CSS/Xpath selector.
      */
-    element(property: string | Element | RelativeBy): ExpectElement;
+    element(property: Definition): ExpectElement;
```

Comment:
I think this should be `Omit<Definition, ElementProperties>`?

Reply:
Not sure if understand you right.

But in my opinion it allows any kind of selector. 
Unfortunately the "ElementProperties" selector is missing so far, even though supported (afaik):
https://github.com/nightwatchjs/nightwatch/blob/c901b925a68f14ac6ff7c2ed7d0e0730fe936a7b/lib/api/expect/element.js#L71